tephra-core/src/main/java/org/apache/tephra/TransactionManager.java

```diff
@@ -348,27 +348,30 @@ private void cleanupTimedOutTransactions() {
         }
 
         long currentTime = System.currentTimeMillis();
-        List<Long> timedOut = Lists.newArrayList();
+        Map<Long, InProgressType> timedOut = Maps.newHashMap();
```

Comment:
How about the "write lock" comment? Currently we are only holding the read lock and modifying the invalid list.